operator/pkg/gateway-api/gateway_reconcile.go

```diff
@@ -35,7 +36,7 @@ import (
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
-func (r *gatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *gatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reconcileErr error) {
```

Comment:
tbh I'm not a big fan of named return params, I generally prefer the increased readability of defining things in the funcction.

In this case, I guess I can see where you're coming from, but I don't like the look of apparently ignoring one of the return values with a `_`.

Could you change to return an empty value instead for the `ctrl.Result`?

Reply:
me neither :smile:  And i don't like handling "businesslogic" (status handling) in `defers` too. Having to use the named return params is the consequence of this :unamused:  

IMO the only proper option would be to handle this explicitly in some utility function.

At first i thought this PR should be about fixing the bug first - without making this refactoring. But having this discussion (:+1:) leads me to introduce a `handleReconcileErrorWithStatus` function that handles this case explicitly.

BTW: This was also the reason why i introduced the named param only for the `error` and leaving the other one unnamed (ignored -  `_`)  - to avoid using the named params outside of the defer statement.

Reply:
refactored. PTAL @youngnick

Reply:
I love the new approach, should we do the same on the other reconcilers?

Reply:
Thanks for the feedback. Sure - will do!

Reply:
@youngnick applied to the other route reconcilers. PTAL